src/Rules/Classes/DuplicateDeclarationRule.php

```diff
@@ -0,0 +1,80 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Classes;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\InClassNode;
+use PHPStan\Rules\RuleErrorBuilder;
+use function array_key_exists;
+use function sprintf;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PHPStan\Node\InClassNode>
+ */
+class DuplicateDeclarationRule implements \PHPStan\Rules\Rule
+{
+
+	public function getNodeType(): string
+	{
+		return InClassNode::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		$classScope = $scope->getClassReflection();
+		if ($classScope === null) {
+			throw new \PHPStan\ShouldNotHappenException();
+		}
+
+		/** @var InClassNode $node */
```

Comment:
Not necessary

Reply:
I added that to get autocomplete in PhpStorm while I was working on the code. I'll get rid of it.